src/guide-seeding.md

```diff
@@ -0,0 +1,125 @@
+
+## Seeding from ...
+
+### Fresh entropy
+
+Using a fresh seed (direct from the OS) is easy using [`SeedableRng::from_entropy`]:
+
+```rust,editable
+use rand::prelude::*;
+use rand_chacha::ChaCha20Rng;
+
+let rng = ChaCha20Rng::from_entropy();
+```
+
+Note that this requires `rand_core` has the feature `getrandom` enabled.
+
+### Another RNG
+
+Quite obviously, another RNG may be used to fill a seed. We provide a
+convenience method for this:
+
+```rust,editable
+use rand::prelude::*;
+use rand_pcg::Pcg64;
+
+let rng = Pcg64::from_rng(thread_rng());
+```
+
+But, say you want to save a key and use it later. For that you need to be a
+little bit more explicit:
+
+```rust,editable
+use rand::prelude::*;
+use rand_chacha::ChaCha8Rng;
+
+let mut seed: <ChaCha8Rng as SeedableRng>::Seed = Default::default();
+thread_rng().fill(&mut seed);
+let rng = ChaCha8Rng::from_seed(seed);
+```
+
+**Obligatory warning**: a few simple PRNGs, notably [`XorShiftRng`],
+behave badly when seeded from the same type of generator (in this case, Xorshift
+generates a clone). For cryptographic PRNGs this is not a problem;
+for others it is recommended to seed from a different type of generator.
+[`ChaCha8Rng`] is an excellent choice for a deterministic master generator
+(but for cryptographic uses, prefer the 12-round variant or higher).
+
+### A simple number
+
+For some applications, especially simulations, all you want are a sequence of
+distinct, fixed random number seeds, e.g. 1, 2, 3, etc.
+
+[`SeedableRng::seed_from_u64`] is designed exactly for this use-case.
+Internally, it uses a simple PRNG to fill the bits of the seed from the input
+number while providing good bit-avalance (so that two similar numbers such as
+0 and 1 translate to very different seeds and independent RNG sequences).
+
+```rust,editable
+use rand::prelude::*;
+use rand_pcg::Pcg64;
+
+let rng = Pcg64::seed_from_u64(2);
+```
+
+Note that a number with 64-bits or less **cannot be secure**, so this should
+not be used for applications such as cryptography or gambling games.
+
+### A string, or any hashable data
+
+Say you let users enter a string to seed the random number generator. Ideally,
+all parts of the string should influence the generator and making only a small
+change to the string should result in a fully independent generator sequence.
+
+This can be achieved via use of a hash function to compress all input data down
+to a hash result, then using that result to seed a generator. The
+[`rand_seeder`] crate is designed for just this purpose.
+
+```rust,editable
+use rand::prelude::*;
+use rand_seeder::{Seeder, SipHasher};
+use rand_pcg::Pcg64;
+
+let rng: Pcg64 = Seeder::from("stripy zebra").make_rng();
+
+// Or, more explicitly:
+let hasher = SipHasher::from("a sailboat");
+let hasher_rng = hasher.into_rng(); // this is a full RNG: use it directly if you want
+let mut seed = [0u8; 16];
+hasher_rng.fill(&mut seed);
```

Comment:
At this point someone would use `next()` or something like `gen_range()` to get their new number?

Reply:
Ah... the comment is saying that one can simply do `hasher_rng.gen_range(1, 7)` (and not seed a new RNG) if you wish.

How should I put this then?

Reply:
Oh, so the `SipHasher::from("a sailboat")` and `let mut seed = [0u8; 16];
hasher_rng.fill(&mut seed);` are just two different ways to seed said RNG? If so I think just making that a little more clear in the comments would be fine.

Otherwise I think I was able to follow this a bit better than the previous examples.

Reply:
The zeros from `[0u8; 16]` don't matter, because they get overwritten by the `hasher_rng.fill` call. (It is not possible to use an uninitialized array without resorting to `unsafe`, which is why it's initialized with zeros in the example.)